### PR TITLE
VZ-2333.  Multicluster acceptance test pipeline for KinD 

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Enables multicluster acceptance testing on KinD in Jenkins.
- install up to 3 clusters (1 admin, to managed) 
- install VZ on each in parallel (using `managed-cluster` profile on the mgd clusters)
- execute the verify-infra tests that are currently working with it.

OKE support will be followed up through a separate PR.

Fixes VZ-2333

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/multicluster/Jenkinsfile [hunk 1]
```diff
@@ -365,6 +476,56 @@ pipeline {
...
+                echo "step 1"
+                export KUBECONFIG=${KUBECONFIG_DIR}/$count/kube_config
+                cd ${GO_REPO_PATH}/verrazzano
+                echo "Installing the Verrazzano Platform Operator"
+                kubectl apply -f ${WORKSPACE}/acceptance-test-operator.yaml
+
+                # make sure ns exists
+                echo "step 2"
+                ./tests/e2e/config/scripts/check_verrazzano_ns_exists.sh verrazzano-install
+
```
Comment (markxnelson) [inline ci/multicluster/Jenkinsfile:510]: i miss step 3

**Review by markxnelson — APPROVED** [review]

File: ci/multicluster/Jenkinsfile [hunk 2]
```diff
@@ -36,15 +35,21 @@ pipeline {
...
-        booleanParam (description: 'Whether to dump k8s cluster on success (off by default can be useful to capture for comparing to failed cluster)', name: 'DUMP_K8S_CLUSTER_ON_SUCCESS', defaultValue: false)
+        choice (name: 'KUBERNETES_CLUSTER_VERSION',
+                description: 'Kubernetes Version for KIND Cluster',
+                // 1st choice is the default value
+                choices: [ "1.18", "1.17" ])
+        string (name: 'VERRAZZANO_OPERATOR_IMAGE',
+                        defaultValue: 'NONE',
+                        description: 'Verrazzano platform operator image name (in ghcr.io repo).  If not specified, the latest operator.yaml published to the Verrazzano Object Store will be used',
+                        trim: true)
+        string name: "ADMIN_CLUSTER_PROFILE", defaultValue: 'dev', description: 'Verrazzano install profile name',  trim: true
```
Comment (mgianatagh) [inline ci/multicluster/Jenkinsfile:50]: Change description to be consistent to managed cluster profile?
'Verrazzano Admin Cluster install profile name'

File: ci/multicluster/Jenkinsfile [hunk 3]
```diff
@@ -261,9 +285,96 @@ pipeline {
...
+                            // Create a dictionary of Verrazzano install steps to be executed in parallel
+                            // - the first one will always be the Admin cluster
+                            // - clusters 2-max are managed clusters
+                            def verrazzanoInstallStages = [:]
+                            verrazzanoInstallStages["vz-admin"] = installVerrazzano(1, "${ADMIN_CLUSTER_INSTALL_CONFIG_FILE}")
+                            int clusterCount = params.TOTAL_CLUSTERS.toInteger()
+                            for(int count=2; count<=clusterCount; count++) {
+                                verrazzanoInstallStages["vz-mgd-${count-1}"] = installVerrazzano(count, "${MANAGED_CLUSTER_INSTALL_CONFIG_FILE}")
+                            }
+                            parallel verrazzanoInstallStages
```
Comment (desagar) [inline ci/multicluster/Jenkinsfile:345]: this one is for the admin cluster I presume? and that's the default profile?

**Review by desagar — APPROVED** [review]

File: ci/multicluster/Jenkinsfile (hunk 1 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:510]: That was just debug stuff.  I need to remove that, I'll do that as part of the next JIRA.

File: ci/multicluster/Jenkinsfile (hunk 3 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:345]: The code above creates a "vz-admin" step outside the loop, and that's always cluster "1".  The remaining slots (2/3) are "vz-mgd-x" steps, set up inside the loop.

The previous KinD stage sets up the admin and managed-cluster VZ install yamls.

File: ci/multicluster/Jenkinsfile [hunk 4]
```diff
@@ -261,9 +285,96 @@ pipeline {
...
+                            sh """
+                                echo "Platform Operator Configuration"
+                                cd ${GO_REPO_PATH}/verrazzano
+                                if [ "NONE" = "${VERRAZZANO_OPERATOR_IMAGE}" ]; then
+                                    echo "Using the latest master branch operator.yaml from object storage"
+                                    oci --region us-phoenix-1 os object get --namespace ${OCI_OS_NAMESPACE} -bn ${OCI_OS_BUCKET} --name master/operator.yaml --file ${WORKSPACE}/downloaded-operator.yaml
+                                    cp ${WORKSPACE}/downloaded-operator.yaml ${WORKSPACE}/acceptance-test-operator.yaml
+                                else
+                                    echo "Generating operator.yaml based on image name provided: ${VERRAZZANO_OPERATOR_IMAGE}"
+                                    ./tests/e2e/config/scripts/process_operator_yaml.sh platform-operator "${VERRAZZANO_OPERATOR_IMAGE}" verrazzano-container-registry ghcr.io ${WORKSPACE}/acceptance-test-operator.yaml
```
Comment (mgianatagh) [inline ci/multicluster/Jenkinsfile:331]: process_operator_yaml.sh script does not exist anymore.  Look at the verrazzano/Jenkinsfile file to see how that is being done now.  There is a make target "generate-operator-yaml" in the platform-operator make file.

File: ci/multicluster/Jenkinsfile (hunk 2 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:50]: Fixed in new branch for the OKE work.

File: ci/multicluster/Jenkinsfile (hunk 4 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:331]: OK, I think I may have lifted this from a version before that went in or something.... I definitely lifted this from there.

Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:331]: actually, I don't think I want to run the `make` command right?  I think that builds a full image and then crates it, I just want to use an existing one specified in the parameters.

Looks like there's another script tool under `tools/scripts/generate_operator_yaml.sh` used in ci/kind/Jenkinsfile

Comment (michael-cico) [reply ci/multicluster/Jenkinsfile:331]: Fixed in the new branch.

**Pull request merged**